File: api/src/main/java/org/junotb/api/registration/web/RegistrationResponse.java
package org.junotb.api.registration.web;

import org.junotb.api.registration.Registration;
import org.junotb.api.registration.RegistrationStatus;

import java.time.OffsetDateTime;

public record RegistrationResponse(
    Long id,
    Long scheduleId,
    String studentId,
    RegistrationStatus status,
    OffsetDateTime registeredAt
) {
    public static RegistrationResponse from(Registration registration) {
        return new RegistrationResponse(
            registration.getId(),
            registration.getSchedule().getId(),
            registration.getStudent().getId(),
            registration.getStatus(),
            registration.getRegisteredAt()
        );
    }
}

File: web/src/schema/registration/registration-response.ts
import { z } from "zod";

export const RegistrationStatusSchema = z.enum(["REGISTERED", "CANCELED"]);
export type RegistrationStatus = z.infer<typeof RegistrationStatusSchema>;

export const RegistrationSchema = z.object({
  id: z.number(),
  scheduleId: z.number(),
  studentId: z.string(),
  status: RegistrationStatusSchema,
  registeredAt: z.string(),
});

export type Registration = z.infer<typeof RegistrationSchema>;
